In [1]:
import pandas as pd

## Calibration set 1

In [18]:
# Load the dataset
df1 = pd.read_excel("TIAB_Calibration1_labeled_corrected.xlsx")
df1

MID                               doi  \
0     M3062                 10.2196/jmir.2460   
1     M7694              10.4088/JCP.13m08910   
2      M902         10.1007/s00038-014-0596-0   
3     M6722                               NaN   
4    M11183              10.4088/JCP.13m08914   
..      ...                               ...   
295  M10364        10.1016/j.nicl.2023.103417   
296  M10348        10.1192/j.eurpsy.2023.1919   
297   M8519  10.1016/j.pharmthera.2023.108431   
298   M3062                 10.2196/jmir.2460   
299   M4096                 10.12740/pp/22256   

                                                 title  \
0    Internet-based early intervention to prevent p...   
1    Postdeployment suicidal ideations and trajecto...   
2    Maternal posttraumatic stress disorder symptom...   
3    Trajectories of PTSD risk and resilience in Wo...   
4    Latent trajectories of trauma symptoms and res...   
..                                                 ...   
295  Altered Resting-State functional connectivity ...   
296  Psychosocial and Physical Rehabilitation of Bu...   
297  The antidepressant actions of ketamine and its...   
298  Internet-based early intervention to prevent p...   
299  Research and treatment of war neuroses at the ...   

                                              abstract  year title_eligible  \
0    BACKGROUND: Posttraumatic stress disorder (PTS...  2013              Y   
1    Objective: Suicidal ideation in veterans is of...  2014              Y   
2    OBJECTIVES: This study examined trajectories o...  2014              Y   
3    Longitudinal symptoms of post-traumatic stress...  2014              Y   
4    Objective: To identify trajectories of posttra...  2014              Y   
..                                                 ...   ...            ...   
295  Background: Post-traumatic stress disorder can...  2023              N   
296  Introduction: Globally, burns are responsible ...  2023              Y   
297  Ketamine, an N-methyl-d-aspartate receptor (NM...  2023              N   
298  BACKGROUND: Posttraumatic stress disorder (PTS...  2013            NaN   
299  The aim of this article is to offer an overvie...  2014            NaN   

    TI-AB_IC1 TI-AB_IC2 TI-AB_IC3 TI-AB_IC4 TI-AB_other_exlusion_reason  \
0           Y         Y         Y         Y                         NaN   
1           Y         Y         Y         Y                         NaN   
2           Y         Q         Y         Y                         NaN   
3           Y         Q         Y         Q                         NaN   
4           Y         Y         Y         Q                         NaN   
..        ...       ...       ...       ...                         ...   
295       NaN       NaN       NaN       NaN                         NaN   
296         N         N         N         N                         NaN   
297       NaN       NaN       NaN       NaN                         NaN   
298       NaN       NaN       NaN       NaN             Same as line 43   
299       NaN       NaN       NaN       NaN                   Duplicate   

     TI-AB_final_label  
0                  1.0  
1                  1.0  
2                  1.0  
3                  1.0  
4                  1.0  
..                 ...  
295                0.0  
296                0.0  
297                0.0  
298                NaN  
299                NaN  

[300 rows x 12 columns]

In [19]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 300 entries, 0 to 299
Data columns (total 12 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   MID                          300 non-null    object 
 1   doi                          262 non-null    object 
 2   title                        300 non-null    object 
 3   abstract                     300 non-null    object 
 4   year                         300 non-null    int64  
 5   title_eligible               298 non-null    object 
 6   TI-AB_IC1                    181 non-null    object 
 7   TI-AB_IC2                    181 non-null    object 
 8   TI-AB_IC3                    181 non-null    object 
 9   TI-AB_IC4                    181 non-null    object 
 10  TI-AB_other_exlusion_reason  14 non-null     object 
 11  TI-AB_final_label            298 non-null    float64
dtypes: float64(1), int64(1), object(10)
memory usage: 28.3+ KB


In [20]:
# Keep only the useful columns for my project
title_df = df1[['MID','title', 'title_eligible']]
title_df

MID                                              title title_eligible
0     M3062  Internet-based early intervention to prevent p...              Y
1     M7694  Postdeployment suicidal ideations and trajecto...              Y
2      M902  Maternal posttraumatic stress disorder symptom...              Y
3     M6722  Trajectories of PTSD risk and resilience in Wo...              Y
4    M11183  Latent trajectories of trauma symptoms and res...              Y
..      ...                                                ...            ...
295  M10364  Altered Resting-State functional connectivity ...              N
296  M10348  Psychosocial and Physical Rehabilitation of Bu...              Y
297   M8519  The antidepressant actions of ketamine and its...              N
298   M3062  Internet-based early intervention to prevent p...            NaN
299   M4096  Research and treatment of war neuroses at the ...            NaN

[300 rows x 3 columns]

In [21]:
# Check how many missing values are present in the dataset
title_df.isna().sum()

MID               0
title             0
title_eligible    2
dtype: int64

There are 2 missing values for the title_eligible column, and 2 missing values for the TI-AB_final_label column. Let's remove them.

In [22]:
title_df = title_df.dropna(subset=['title_eligible'])
title_df

MID                                              title title_eligible
0     M3062  Internet-based early intervention to prevent p...              Y
1     M7694  Postdeployment suicidal ideations and trajecto...              Y
2      M902  Maternal posttraumatic stress disorder symptom...              Y
3     M6722  Trajectories of PTSD risk and resilience in Wo...              Y
4    M11183  Latent trajectories of trauma symptoms and res...              Y
..      ...                                                ...            ...
293  M10369  Time-course analysis of frontal gene expressio...              N
294   M6107  Treatment outcomes of Veteran men in a compreh...              Y
295  M10364  Altered Resting-State functional connectivity ...              N
296  M10348  Psychosocial and Physical Rehabilitation of Bu...              Y
297   M8519  The antidepressant actions of ketamine and its...              N

[298 rows x 3 columns]

In [23]:
title_df.isna().sum()

MID               0
title             0
title_eligible    0
dtype: int64

Now there are no missing values anymore. The total rows are now 298.

In [24]:
title_df['title_eligible'].value_counts()

title_eligible
Y    179
N    119
Name: count, dtype: int64

In [25]:
# Convert the Y and N with 1 and 0 labels in "title_eligible_Bruno"
title_df['title_eligible'] = title_df['title_eligible'].apply(lambda x: 0 if x=='N' else 1)
title_df

C:\Users\giuli\AppData\Local\Temp\ipykernel_19952\2311393162.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  title_df['title_eligible'] = title_df['title_eligible'].apply(lambda x: 0 if x=='N' else 1)


MID                                              title  title_eligible
0     M3062  Internet-based early intervention to prevent p...               1
1     M7694  Postdeployment suicidal ideations and trajecto...               1
2      M902  Maternal posttraumatic stress disorder symptom...               1
3     M6722  Trajectories of PTSD risk and resilience in Wo...               1
4    M11183  Latent trajectories of trauma symptoms and res...               1
..      ...                                                ...             ...
293  M10369  Time-course analysis of frontal gene expressio...               0
294   M6107  Treatment outcomes of Veteran men in a compreh...               1
295  M10364  Altered Resting-State functional connectivity ...               0
296  M10348  Psychosocial and Physical Rehabilitation of Bu...               1
297   M8519  The antidepressant actions of ketamine and its...               0

[298 rows x 3 columns]

In [26]:
# Rename eligibility column and hiding the abstract column for Title Simulation study
ti_sim_df = title_df.rename(columns={'title_eligible': 'label'})
ti_sim_df

MID                                              title  label
0     M3062  Internet-based early intervention to prevent p...      1
1     M7694  Postdeployment suicidal ideations and trajecto...      1
2      M902  Maternal posttraumatic stress disorder symptom...      1
3     M6722  Trajectories of PTSD risk and resilience in Wo...      1
4    M11183  Latent trajectories of trauma symptoms and res...      1
..      ...                                                ...    ...
293  M10369  Time-course analysis of frontal gene expressio...      0
294   M6107  Treatment outcomes of Veteran men in a compreh...      1
295  M10364  Altered Resting-State functional connectivity ...      0
296  M10348  Psychosocial and Physical Rehabilitation of Bu...      1
297   M8519  The antidepressant actions of ketamine and its...      0

[298 rows x 3 columns]

In [32]:
# Save the dataset for the title simulation
ti_sim_df.to_excel("Title_Simulation.xlsx")

In [33]:
# Make a test dataset for the LLM without the title labels
test_df = ti_sim_df.drop(['label'], axis='columns')
test_df

MID                                              title
0     M3062  Internet-based early intervention to prevent p...
1     M7694  Postdeployment suicidal ideations and trajecto...
2      M902  Maternal posttraumatic stress disorder symptom...
3     M6722  Trajectories of PTSD risk and resilience in Wo...
4    M11183  Latent trajectories of trauma symptoms and res...
..      ...                                                ...
293  M10369  Time-course analysis of frontal gene expressio...
294   M6107  Treatment outcomes of Veteran men in a compreh...
295  M10364  Altered Resting-State functional connectivity ...
296  M10348  Psychosocial and Physical Rehabilitation of Bu...
297   M8519  The antidepressant actions of ketamine and its...

[298 rows x 2 columns]

In [34]:
# Save the test dataset
test_df.to_excel("Test_LLM.xlsx")

In [27]:
# Make a dataset with the abstract labels
abs_df = df1[['MID','title', 'TI-AB_final_label']]
abs_df

MID                                              title  \
0     M3062  Internet-based early intervention to prevent p...   
1     M7694  Postdeployment suicidal ideations and trajecto...   
2      M902  Maternal posttraumatic stress disorder symptom...   
3     M6722  Trajectories of PTSD risk and resilience in Wo...   
4    M11183  Latent trajectories of trauma symptoms and res...   
..      ...                                                ...   
295  M10364  Altered Resting-State functional connectivity ...   
296  M10348  Psychosocial and Physical Rehabilitation of Bu...   
297   M8519  The antidepressant actions of ketamine and its...   
298   M3062  Internet-based early intervention to prevent p...   
299   M4096  Research and treatment of war neuroses at the ...   

     TI-AB_final_label  
0                  1.0  
1                  1.0  
2                  1.0  
3                  1.0  
4                  1.0  
..                 ...  
295                0.0  
296                0.0  
297                0.0  
298                NaN  
299                NaN  

[300 rows x 3 columns]

In [28]:
abs_df = abs_df.dropna(subset=['TI-AB_final_label'])
abs_df

MID                                              title  \
0     M3062  Internet-based early intervention to prevent p...   
1     M7694  Postdeployment suicidal ideations and trajecto...   
2      M902  Maternal posttraumatic stress disorder symptom...   
3     M6722  Trajectories of PTSD risk and resilience in Wo...   
4    M11183  Latent trajectories of trauma symptoms and res...   
..      ...                                                ...   
293  M10369  Time-course analysis of frontal gene expressio...   
294   M6107  Treatment outcomes of Veteran men in a compreh...   
295  M10364  Altered Resting-State functional connectivity ...   
296  M10348  Psychosocial and Physical Rehabilitation of Bu...   
297   M8519  The antidepressant actions of ketamine and its...   

     TI-AB_final_label  
0                  1.0  
1                  1.0  
2                  1.0  
3                  1.0  
4                  1.0  
..                 ...  
293                0.0  
294                0.0  
295                0.0  
296                0.0  
297                0.0  

[298 rows x 3 columns]

In [29]:
abs_df = abs_df.rename(columns={'TI-AB_final_label': 'label'})
abs_df

MID                                              title  label
0     M3062  Internet-based early intervention to prevent p...    1.0
1     M7694  Postdeployment suicidal ideations and trajecto...    1.0
2      M902  Maternal posttraumatic stress disorder symptom...    1.0
3     M6722  Trajectories of PTSD risk and resilience in Wo...    1.0
4    M11183  Latent trajectories of trauma symptoms and res...    1.0
..      ...                                                ...    ...
293  M10369  Time-course analysis of frontal gene expressio...    0.0
294   M6107  Treatment outcomes of Veteran men in a compreh...    0.0
295  M10364  Altered Resting-State functional connectivity ...    0.0
296  M10348  Psychosocial and Physical Rehabilitation of Bu...    0.0
297   M8519  The antidepressant actions of ketamine and its...    0.0

[298 rows x 3 columns]

In [15]:
abs_df.to_excel("Abstract_Simulation.xlsx")

## Calibration set 2

In [8]:
# Load the dataset
df2 = pd.read_excel("TIAB_Calibration2_labeled.xlsx")
df2

MID                            doi  \
0    M10516           10.1038/npp.2014.282   
1     M7694           10.4088/JCP.13m08910   
2    M11183           10.4088/JCP.13m08914   
3     M6722                            NaN   
4     M6707                            NaN   
..      ...                            ...   
295  M11638             10.1111/jcpp.13835   
296  M11666     10.1186/s12978-023-01604-0   
297   M7102  10.1080/20008066.2023.2172256   
298   M9416         10.4088/JCP.14com09372   
299   M7197               10.1037/a0039077   

                                                 title  \
0    Pre-, peri-, and post-deployment trajectories ...   
1    Postdeployment suicidal ideations and trajecto...   
2    Latent trajectories of trauma symptoms and res...   
3    Trajectories of PTSD risk and resilience in Wo...   
4    Factor predicting maternal posttraumatic stres...   
..                                                 ...   
295  Variation in sleep profiles in children with A...   
296  Overlap between birth trauma and mistreatment:...   
297  Cross-cultural validity and psychometric prope...   
298  Empirical characterization of heterogeneous po...   
299  Cumulative burden of lifetime adversities: Tra...   

                                              abstract  publication_year  \
0    Background: The burden of post traumatic stres...              2014   
1    Objective: Suicidal ideation in veterans is of...              2014   
2    Objective: To identify trajectories of posttra...              2014   
3    Longitudinal symptoms of post-traumatic stress...              2014   
4    Natural disasters are sudden, large-scale even...              2022   
..                                                 ...               ...   
295  Background Sleep difficulties are common in ch...              2023   
296  Plain language summaryMany people giving birth...              2023   
297  Background: The ICD-11 post-traumatic stress d...              2023   
298                                                NaN              2014   
299  This study examined the utility of a lifetime ...              2015   

    title_eligible TI-AB_IC1 TI-AB_IC2 TI-AB_IC3 TI-AB_IC4  \
0                Y         Y         Y         Y         Q   
1                Y         Y         Y         Y         Y   
2                Y         Y         Y         Y         Q   
3                Y         Y         Q         Y         Q   
4                Y         Y         Q         Y         Y   
..             ...       ...       ...       ...       ...   
295              N       NaN       NaN       NaN       NaN   
296              N       NaN       NaN       NaN       NaN   
297              N       NaN       NaN       NaN       NaN   
298            NaN       NaN       NaN       NaN       NaN   
299            NaN       NaN       NaN       NaN       NaN   

                     TI-AB_other_exlusion_reason  TI-AB_final_label  \
0                                            NaN                1.0   
1                                            NaN                1.0   
2                                            NaN                1.0   
3                                            NaN                1.0   
4                                            NaN                1.0   
..                                           ...                ...   
295                                          NaN                0.0   
296                                          NaN                0.0   
297                                          NaN                0.0   
298  Abstract is missing. Record is a commentary                NaN   
299                      Repeated record line 94                NaN   

    Unnamed: 12  
0           NaN  
1           NaN  
2           NaN  
3           NaN  
4           NaN  
..          ...  
295         NaN  
296         NaN  
297         NaN  
298         NaN  
299         NaN  

[300 rows x 13 columns]

In [9]:
df2 = df2.dropna(subset=['TI-AB_final_label'])

In [10]:
title_df = df2[['MID','title', 'title_eligible']]
title_df

MID                                              title title_eligible
0    M10516  Pre-, peri-, and post-deployment trajectories ...              Y
1     M7694  Postdeployment suicidal ideations and trajecto...              Y
2    M11183  Latent trajectories of trauma symptoms and res...              Y
3     M6722  Trajectories of PTSD risk and resilience in Wo...              Y
4     M6707  Factor predicting maternal posttraumatic stres...              Y
..      ...                                                ...            ...
293   M9302  Habituation or sensitization of brain response...              N
294  M11639  Actigraphic evidence of persistent sleep disru...              N
295  M11638  Variation in sleep profiles in children with A...              N
296  M11666  Overlap between birth trauma and mistreatment:...              N
297   M7102  Cross-cultural validity and psychometric prope...              N

[295 rows x 3 columns]

In [11]:
# Check how many missing values are present in the dataset
title_df.isna().sum()

MID               0
title             0
title_eligible    0
dtype: int64

In [12]:
title_df['title_eligible'].value_counts()

title_eligible
N    156
Y    139
Name: count, dtype: int64

In [13]:
# Convert the Y and N with 1 and 0 labels in "title_eligible_Bruno"
title_df['title_eligible'] = title_df['title_eligible'].apply(lambda x: 0 if x=='N' else 1)
title_df

C:\Users\giuli\AppData\Local\Temp\ipykernel_19952\2311393162.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  title_df['title_eligible'] = title_df['title_eligible'].apply(lambda x: 0 if x=='N' else 1)


MID                                              title  title_eligible
0    M10516  Pre-, peri-, and post-deployment trajectories ...               1
1     M7694  Postdeployment suicidal ideations and trajecto...               1
2    M11183  Latent trajectories of trauma symptoms and res...               1
3     M6722  Trajectories of PTSD risk and resilience in Wo...               1
4     M6707  Factor predicting maternal posttraumatic stres...               1
..      ...                                                ...             ...
293   M9302  Habituation or sensitization of brain response...               0
294  M11639  Actigraphic evidence of persistent sleep disru...               0
295  M11638  Variation in sleep profiles in children with A...               0
296  M11666  Overlap between birth trauma and mistreatment:...               0
297   M7102  Cross-cultural validity and psychometric prope...               0

[295 rows x 3 columns]

In [14]:
# Rename eligibility column and hiding the abstract column for Title Simulation study
ti_val_df = title_df.rename(columns={'title_eligible': 'label'})
ti_val_df

MID                                              title  label
0    M10516  Pre-, peri-, and post-deployment trajectories ...      1
1     M7694  Postdeployment suicidal ideations and trajecto...      1
2    M11183  Latent trajectories of trauma symptoms and res...      1
3     M6722  Trajectories of PTSD risk and resilience in Wo...      1
4     M6707  Factor predicting maternal posttraumatic stres...      1
..      ...                                                ...    ...
293   M9302  Habituation or sensitization of brain response...      0
294  M11639  Actigraphic evidence of persistent sleep disru...      0
295  M11638  Variation in sleep profiles in children with A...      0
296  M11666  Overlap between birth trauma and mistreatment:...      0
297   M7102  Cross-cultural validity and psychometric prope...      0

[295 rows x 3 columns]

In [46]:
# Save the dataset for the title simulation
ti_val_df.to_excel("Title_Validation.xlsx")

In [47]:
# Make a test dataset for the LLM without the title labels
test_df = ti_val_df.drop(['label'], axis='columns')
test_df

MID                                              title
0    M10516  Pre-, peri-, and post-deployment trajectories ...
1     M7694  Postdeployment suicidal ideations and trajecto...
2    M11183  Latent trajectories of trauma symptoms and res...
3     M6722  Trajectories of PTSD risk and resilience in Wo...
4     M6707  Factor predicting maternal posttraumatic stres...
..      ...                                                ...
293   M9302  Habituation or sensitization of brain response...
294  M11639  Actigraphic evidence of persistent sleep disru...
295  M11638  Variation in sleep profiles in children with A...
296  M11666  Overlap between birth trauma and mistreatment:...
297   M7102  Cross-cultural validity and psychometric prope...

[295 rows x 2 columns]

In [48]:
# Save the test dataset
test_df.to_excel("Validation_LLM.xlsx")

In [15]:
# Make a dataset with the abstract labels
abs_df = df2[['MID','title', 'TI-AB_final_label']]
abs_df

MID                                              title  \
0    M10516  Pre-, peri-, and post-deployment trajectories ...   
1     M7694  Postdeployment suicidal ideations and trajecto...   
2    M11183  Latent trajectories of trauma symptoms and res...   
3     M6722  Trajectories of PTSD risk and resilience in Wo...   
4     M6707  Factor predicting maternal posttraumatic stres...   
..      ...                                                ...   
293   M9302  Habituation or sensitization of brain response...   
294  M11639  Actigraphic evidence of persistent sleep disru...   
295  M11638  Variation in sleep profiles in children with A...   
296  M11666  Overlap between birth trauma and mistreatment:...   
297   M7102  Cross-cultural validity and psychometric prope...   

     TI-AB_final_label  
0                  1.0  
1                  1.0  
2                  1.0  
3                  1.0  
4                  1.0  
..                 ...  
293                0.0  
294                0.0  
295                0.0  
296                0.0  
297                0.0  

[295 rows x 3 columns]

In [16]:
# Check how many missing values are present in the dataset
abs_df.isna().sum()

MID                  0
title                0
TI-AB_final_label    0
dtype: int64

In [17]:
abs_df = abs_df.rename(columns={'TI-AB_final_label': 'label'})
abs_df

MID                                              title  label
0    M10516  Pre-, peri-, and post-deployment trajectories ...    1.0
1     M7694  Postdeployment suicidal ideations and trajecto...    1.0
2    M11183  Latent trajectories of trauma symptoms and res...    1.0
3     M6722  Trajectories of PTSD risk and resilience in Wo...    1.0
4     M6707  Factor predicting maternal posttraumatic stres...    1.0
..      ...                                                ...    ...
293   M9302  Habituation or sensitization of brain response...    0.0
294  M11639  Actigraphic evidence of persistent sleep disru...    0.0
295  M11638  Variation in sleep profiles in children with A...    0.0
296  M11666  Overlap between birth trauma and mistreatment:...    0.0
297   M7102  Cross-cultural validity and psychometric prope...    0.0

[295 rows x 3 columns]

In [52]:
abs_df.to_excel("Abstract_Validation.xlsx")